beacon_node/beacon_chain/tests/sync_committee_verification.rs

```diff
@@ -0,0 +1,668 @@
+    harness.add_attested_blocks_at_slots(
+        state,
+        Hash256::zero(),
+        &[Slot::new(1), Slot::new(2)],
+        (0..VALIDATOR_COUNT).collect::<Vec<_>>().as_slice(),
+    );
+
+    let current_slot = harness.chain.slot().expect("should get slot");
+
+    let (valid_aggregate, aggregator_index, aggregator_sk) =
+        get_valid_sync_contribution(&harness, RelativeSyncCommittee::Current);
+
+    macro_rules! assert_invalid {
+            ($desc: tt, $attn_getter: expr, $($error: pat) |+ $( if $guard: expr )?) => {
+                assert!(
+                    matches!(
+                        harness
+                            .chain
+                            .verify_sync_contribution_for_gossip($attn_getter)
+                            .err()
+                            .expect(&format!(
+                                "{} should error during verify_sync_contribution_for_gossip",
+                                $desc
+                            )),
+                        $( $error ) |+ $( if $guard )?
+                    ),
+                    "case: {}",
+                    $desc,
+                );
+            };
+        }
+
+    /*
+     * The following two tests ensure:
+     *
+     * The contribution's slot is for the current slot, i.e. contribution.slot == current_slot
+     * (with a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance).
+     */
+
+    let future_slot = current_slot + 1;
+    assert_invalid!(
+        "aggregate from future slot",
+        {
+            let mut a = valid_aggregate.clone();
+            a.message.contribution.slot = future_slot;
+            a
+        },
+        SyncCommitteeError::FutureSlot { message_slot, latest_permissible_slot }
+        if message_slot == future_slot && latest_permissible_slot == current_slot
+    );
+
+    let early_slot = current_slot
+        .as_u64()
+        // Subtract an additional slot since the harness will be exactly on the start of the
+        // slot and the propagation tolerance will allow an extra slot.
+        .checked_sub(2)
+        .expect("chain is not sufficiently deep for test")
+        .into();
+    assert_invalid!(
+        "aggregate from past slot",
+        {
+            let mut a = valid_aggregate.clone();
+            a.message.contribution.slot = early_slot;
+            a
+        },
+        SyncCommitteeError::PastSlot {
+            message_slot,
+
+            earliest_permissible_slot
+        }
+        if message_slot == early_slot
+            && earliest_permissible_slot == current_slot - 1
+    );
+
+    /*
+     * The following test ensures:
+     *
+     * The subcommittee index is in the allowed range,
+     * i.e. `contribution.subcommittee_index < SYNC_COMMITTEE_SUBNET_COUNT`.
+     */
+
+    assert_invalid!(
+        "subcommittee index out of range",
+        {
+            let mut a = valid_aggregate.clone();
+            a.message.contribution.subcommittee_index = SYNC_COMMITTEE_SUBNET_COUNT + 1;
```

Comment:
Can we set this to `SYNC_COMMITTEE_SUBNET_COUNT` exactly so that we guard against an off-by-one in the check?